### PR TITLE
Allow multiple reasons for a payload reject in script and UI

### PR DESCRIPTION
**Pull request by DennisPeriquet**
[TRT-1186](https://issues.redhat.com//browse/TRT-1186)

TODO:

- [x] Test DB migration to introduce the new `reject_reasons` column (test locally; repeat with `ALTER TABLE release_tags; DROP COLUMN reject_reasons;`)
- [x] Test `rejected-payloads.py` script:
  - [x] ensure it can read the new `reject_reasons` column (using `list` subcommand)
  - [x] ensure it can write to the new `reject_reasons` column (using `categorize` subcommand)
- [x] Ensure the sippy UI displays correctly
  - [x] handle case for no reject_reason
  - [x] handle case for reject_reason present
  - [x] handle case for reject_reasons with 1 element
  - [x] handle case for reject_reasons with multiple elements
  - [x] handle case for reject_reasons greater than 3 elements

sample output for multiple reasons:

```
$ ./rejected-payloads.py -d $DSN --days 4 list -s nightly -r 4.12 --all
index     release tag                                       phase               reject reasons      note
1         4.12.0-0.nightly-2023-08-08-180643                Rejected            TEST_FLAKE          test flakes are flaky
2         4.12.0-0.nightly-2023-08-08-180641                Rejected            TEST_FLAKE          this is multiple lines
                                                                                PRODUCT_REGRESSION  
                                                                                TEST_REGRESSION     
                                                                                CLOUD_QUOTA         
```

Screenshot 2023-08-10 at 3 48 55 PM

### CODE REVIEW

File: scripts/rejected-payloads.py [hunk 1]
```diff
@@ -98,14 +99,19 @@ def categorizeSingle(session, tag):
...
             try:
-                index = int(val)
-                if index > 0 and index <= len(reject_reasons_keys):
+                # Values outside the legal range are ignored.  Invalid values throw an exception so you can
+                # try again.
+                selected_indexes = [i for i in map(int, val.split()) if 1 <= i <= len(reject_reasons_keys)]
+                if selected_indexes:
                     break
             except ValueError:
+                print(f"  This contains invalid integer values: {val}")
```
Comment (DennisPeriquet) [inline scripts/rejected-payloads.py:110]: "f strings" is a python3 feature -- if we want python2, I can change this to use the  percent method.

File: sippy-ng/src/releases/ReleasePayloadTable.js [hunk 2]
```diff
@@ -124,13 +124,43 @@ function ReleasePayloadTable(props) {
...
+        let display_reasons = []
+
+        // Until we migrate to just reject_reasons, we'll display the current value
+        // of reject_reason if present.
+        if (params.row.reject_reason) {
+          display_reasons.push(params.row.reject_reason)
+        }
+
+        // Since the reject_reason will be the first element, we only need anything after that.
+        if (params.row.reject_reasons && params.row.reject_reasons.length > 1) {
```
Comment (deepsm007) [inline sippy-ng/src/releases/ReleasePayloadTable.js:140]: the field used is `reject_reason`. Either switch the field to `reject_reasons` to accomomdate this changes

Comment (DennisPeriquet) [reply sippy-ng/src/releases/ReleasePayloadTable.js:140]: For now, we still need `reject_reasons` for the previous payloads (the ones before my change is merged).  Although I look for `reject_reason` above, the row returned from the api contains all fields including `reject_reason` and `reject_reasons` so I'm able to consider both when determining what to display.

File: scripts/rejected-payloads.md [hunk 3]
```diff
@@ -81,7 +86,8 @@ Additional options can be provided to limit the scope of the query.
 ```
   -r RELEASE, --release RELEASE    Specifies a release, like 4.11
   -s STREAM, --stream STREAM       Specifies a stream, like nightly or ci
-  -a, --all                        List all rejected payloads. If not specified , list only uncategorized ones.
+  -a ARCH, --arch ARCH             Specifies an architecture, like amd64
+  --all                            List all rejected payloads. If not specified , list only uncategorized ones.
```
[the remark this answers is not in the document]
Comment (DennisPeriquet) [reply scripts/rejected-payloads.md:90]: Agreed.  Let me see about getting rid of that in the README.md

File: scripts/rejected-payloads.md [hunk 4]
```diff
@@ -70,20 +67,14 @@ Please choose the reject reason for tag 4.11.0-0.nightly-2022-06-28-111405 from
...
 
 List command can be used to list uncategorized release tags.
 Additional options can be provided to limit the scope of the query.
 
-```
-  -r RELEASE, --release RELEASE    Specifies a release, like 4.11
-  -s STREAM, --stream STREAM       Specifies a stream, like nightly or ci
-  -a, --all                        List all rejected payloads. If not specified , list only uncategorized ones.
-```
-
```
Comment (dgoodwin) [inline scripts/rejected-payloads.md:86]: Could you add -a amd64 to all the commands in here.

**Pull request merged**